src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+
+                # We correct the computation of y_np1 using y_np1
+                # as part of our prediction Harier equation 5.7
+
+                y_np1 = y_np1 + h*g[steporder+1]*ϕ[3,steporder+1]
+                # TODO: figure out why following does not work instead
+                # for i=1:length(y_np1)
+                #     y_np1[i] += h*g[steporder+1]*ϕstar[3,steporder+1][i]
+                # end
+
+                # (E)valuate estimate for next derivative again
+                dy_np1=F(t_np1,y_np1)
+            end
+            #---################################################################
+            # Error Check for step size `h` and Variable next step selection:
+            #     Check error to determine if step size should be accepted:
+            #     ->if yes, compute `next_h` for next step
+            #     ->if no, use this `next_h`` for next attempt at this step
+            #---################################################################
+
+            #Don't use adaptive steps until n >= steporder + 2
+            if n==1
+                #TODO: adaptive step selection for n<steporder +2
+                successful_step = true
+                next_h = h/4
+            elseif n<steporder+2
+                yerr_k = h*(g[steporder+1] - g[steporder])*ϕ[3,steporder+1]
+                err_k, next_h,timeout_VS = stepsize_hw92!(h, tdir, y[n], y_np1, yerr_k, steporder,
+                                timeout_VS, dof, abstol, reltol, maxstep, norm)
+                if err_k <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    h = fac*next_h # make sure fac < 1
+                end
+            else
+                # We calculate local truncation error, LE_{k+1}; the difference
+                # between implicit results for order k and k+1.
+
+                # We need ϕ_{k+1}(n+1) and g_{k+1}(n);the latter has already
+                # been calculated above, the former still needs calculation
+                ϕ_coefs!(ϕ, ϕstar,dy_np1,steporder+2)
+
+                # Calculate ||LE_{k+1}|| and next stepsize
+                yerr_kp1 = h*(g[steporder+2] - g[steporder+1])*ϕ[3,steporder+2]
+                err_kp1, next_h, timeout_VS = stepsize_hw92!(h,tdir,y[n],y_np1,
+                                            yerr_kp1, steporder, timeout_VS,dof,
+                                            abstol, reltol, maxstep, norm)
+
+                # Harier equation 7.4: If ||LE_{k+1}|| =< 1
+                # then step size is accepted and use next_h for next step
+                # Else, then step size is rejected, reattempt with fac*next_h
+                if err_kp1 <= 1
+                    successful_step = true
+                else
+                    successful_step = false
+                    h = fac*next_h # make sure fac < 1
+                end
+            end # End of error check for current step size h
```

Comment:
I think you ought to resize `h_next` to fit into tspan here, instead of https://github.com/JuliaLang/ODE.jl/pull/106/files#diff-43ccd017e404a50f115b60e06e7f1ac5R260